pkg/manager/manager.go

```diff
@@ -160,6 +190,21 @@ func (k *KubeMacPoolManager) waitForSignal() {
 	close(k.restartChannel)
 }
 
+func (k *KubeMacPoolManager) markPodAsLeader() error {
```

Comment:
Can a race happen here? I'm not too familiar with the leader mechanism. Could it happen that two pods would claim to be leaders at the same moment?

Reply:
It should not append this is how we do the leader election in kubevirt and also implemented by kubernetes